File: backend/internal/models/connection.go
package models

import "time"

// ConnectionConfig represents the configuration for a database connection
type ConnectionConfig struct {
	ID       string `json:"id,omitempty"`
	Name     string `json:"name"`
	Type     string `json:"type"` // mysql, postgres, mongodb, redis, sqlite, oracle, sqlserver
	Host     string `json:"host"`
	Port     int    `json:"port"`
	Username string `json:"username"`
	Password string `json:"password"`
	Database string `json:"database"`
	SSL      bool   `json:"ssl,omitempty"`
}

// Connection represents an active database connection
type Connection struct {
	ConnectionConfig
	Status          string     `json:"status"` // connected, disconnected, connecting, error
	CreatedAt       time.Time  `json:"createdAt"`
	LastConnectedAt *time.Time `json:"lastConnectedAt,omitempty"`
	Error           string     `json:"error,omitempty"`
}

// QueryRequest represents a query execution request
type QueryRequest struct {
	ConnectionID string `json:"connection_id"`
	SQL          string `json:"sql"`
}

// DocumentFindRequest represents a MongoDB find request
type DocumentFindRequest struct {
	ConnectionID string                 `json:"connection_id"`
	Database     string                 `json:"database"`
	Collection   string                 `json:"collection"`
	Filter       map[string]interface{} `json:"filter"`
}

File: packages/ui/src/types/connection.ts
import type { DatabaseType } from "./database";

export interface ConnectionConfig {
  id?: string;
  name: string;
  type: DatabaseType;
  host: string;
  port: number;
  username: string;
  password: string;
  database: string;
  ssl?: boolean;
}

export interface Connection extends ConnectionConfig {
  id: string;
  status: "connected" | "disconnected" | "connecting" | "error";
  createdAt: string;
  lastConnectedAt?: string;
  error?: string;
}

export const DEFAULT_PORTS: Record<DatabaseType, number> = {
  mysql: 3306,
  postgres: 5432,
  mongodb: 27017,
  redis: 6379,
  sqlite: 0,
  oracle: 1521,
  sqlserver: 1433,
};
